src/ios/JXcoreExtension.m

```diff
@@ -28,13 +28,118 @@
 #import "JXcoreExtension.h"
 #import "THEAppContext.h"
 
+#import "JXcore.h"
+
 // JXcoreExtension implementation.
 @implementation JXcoreExtension
 
 // Defines methods.
 - (void)defineMethods
 {
-    [[THEAppContext singleton] defineJavaScriptExtensions];
+  THEAppContext *theApp = [THEAppContext singleton];
+
+  // Export the public API to node
+
+  // StartBroadcasting
+  [JXcore addNativeBlock:^(NSArray * params, NSString * callbackId) 
+  {
+    NSLog(@"jxcore: startBroadcasting");
+
+    if ([params count] != 3 || ![params[0] isKindOfClass:[NSString class]] || 
+        ![params[1] isKindOfClass:[NSNumber class]])
+    {
+      NSLog(@"jxcore: startBroadcasting: badParam");
+      [JXcore callEventCallback:callbackId withParams:@[@"Bad argument"]];
```

Comment:
Non Blocking - Wouldn't it be easier to just put in a return here rather than do all the if nesting? In this case the if nesting isn't too bad so it isn't the end of the world but still.